### PR TITLE
Fix 'image' tag isn't visible #1759

### PR DESCRIPTION
**Pull request by Dima-1**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/search/QuickSearchDialogFragment.java [hunk 1]
```diff
@@ -1835,6 +1835,7 @@ public void replaceQueryWithUiFilter(PoiUIFilter filter, String nameFilter) {
 			poiFilterApplied = true;
 			SearchResult sr = searchPhrase.getLastSelectedWord().getResult();
 			sr.object = filter;
+			sr.acceptPrivate = filter.isShowPrivateNeeded();
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/search/QuickSearchDialogFragment.java:1838]: SearchResult is a wrong place to pass parameters to Search

File: OsmAnd-java/src/main/java/net/osmand/data/Amenity.java [hunk 2]
```diff
@@ -427,6 +427,10 @@ public boolean isClosed() {
 		return OSM_DELETE_VALUE.equals(getAdditionalInfo(OSM_DELETE_TAG));
 	}
 
+	public boolean isPrivateEntertainment() {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/data/Amenity.java:430]: Should be isPrivate not specific to entertainment

File: OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java [hunk 3]
```diff
@@ -1052,7 +1051,9 @@ public boolean search(final SearchPhrase phrase, final SearchResultMatcher resul
 			String nameFilter = null;
 			int countExtraWords = 0;
 			Set<String> poiAdditionals = new LinkedHashSet<>();
+			boolean acceptPrivate = false;
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java:1054]: SearchPhrase should have property to accept private or not

File: OsmAnd/src/net/osmand/plus/views/layers/POIMapLayer.java [hunk 4]
```diff
@@ -163,7 +163,7 @@ protected List<Amenity> calculateResult(@NonNull QuadRect latLonBounds, int zoom
 
 								@Override
 								public boolean publish(Amenity object) {
-									return true;
+									return !object.isPrivateEntertainment() || (object.isPrivateEntertainment() && filter.isShowPrivateNeeded());
```
[the remark this answers is not in the document]
Comment (Dima-1) [reply OsmAnd/src/net/osmand/plus/views/layers/POIMapLayer.java]: check removed

**Pull request merged**